app/services/imo_api_service.rb

```diff
@@ -32,7 +27,35 @@ def invite
       action: "Click here"
     )
     response = execute_post(url, body: request_body)
-    process_response(response, url)
+    result = process_response(response, url, "invitation")
+
+    return if result.nil?
+
+    ImoAuthorization.create!(patient: patient, status: result, last_invited_at: Time.current)
+  end
+
+  def send_notification(patient, message)
```

Comment:
I'm not currently guarding against the possibility that the patient doesn't have an ImoAuthorization. If they don't, we'll just get an error response from Imo and it will log to sentry, which will let us know that there may be a problem. The same thing is happening in the invite endpoint. I don't guard against the possibility that they've already been invited. If they have, we'll get an error response of some sort.

Reply:
If someone has already been invited, will we continually attempt to re-invite them based on some sort of schedule? Should we be capturing invitation attempts on our end so we don't keep retrying patients who are not going to opt-in?

Reply:
We are capturing that information in ImoAuthorization, which has a `last_invited_at` attribute.

Reply:
ah okay, great.